settings.php

```diff
@@ -49,10 +49,17 @@
         get_string('loadonvisibledesc', 'filter_imageopt'), 5, $choices));
 
     $choices = [
-        image::WIDTHATTPRSERVELTMAX => get_string('widthattpreserveltmax', 'filter_imageopt'),
-        image::WIDTHATTPRESERVE => get_string('widthattpreserve', 'filter_imageopt')
+        local::WIDTH_ATT_PRESERVE_MAX => get_string('widthattpreserveltmax', 'filter_imageopt'),
+        local::WIDTH_ATT_PRESERVE => get_string('widthattpreserve', 'filter_imageopt')
     ];
     $settings->add(new admin_setting_configselect('filter_imageopt/widthattribute',
         get_string('widthattribute', 'filter_imageopt'),
-        get_string('widthattributedesc', 'filter_imageopt'), image::WIDTHATTPRSERVELTMAX, $choices));
+        get_string('widthattributedesc', 'filter_imageopt'), local::WIDTH_ATT_PRESERVE_MAX, $choices));
+
+    $settings->add(new admin_setting_configtext(
```

Comment:
Now we have the cache for the min dupes we also need to purge the cache when this setting changes - see:

$setting->set_updatedcallback

Reply:
Ah, I just noticed that the cache is keyed by the minimum dupes setting so you can ignore the above comment.